server/src/main/java/org/tctalent/server/model/db/LinkPreview.java

```diff
@@ -17,44 +17,133 @@
 package org.tctalent.server.model.db;
 
 import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
 import javax.persistence.JoinColumn;
 import javax.persistence.ManyToOne;
 import javax.persistence.SequenceGenerator;
 import javax.persistence.Table;
 import javax.validation.constraints.NotBlank;
 import lombok.Getter;
 import lombok.Setter;
+import org.springframework.data.annotation.Id;
 import org.springframework.lang.Nullable;
 
+/**
+ * Displays a preview of the content navigable to via link-formatted text in the Chat Post editor
+ * or sent Chat Post. A good overview of what constitutes a typical link preview and the methodology
+ * used to put ours together can be found in
+ * <a href="https://andrejgajdos.com/how-to-create-a-link-preview/">this blog post</a> by Andrej Gaydos.
+ * All the desired elements aren't necessarily unearthed by our scraping methods, or present in the
+ * first place. The minimal acceptable elements for display on the TC are a valid URL, which we
+ * parse to also provide a domain.
+ */
 @Getter
 @Setter
 @Entity
 @Table(name = "link_preview")
 @SequenceGenerator(name = "seq_gen", sequenceName = "link_preview_id_seq", allocationSize = 1)
 public class LinkPreview extends AbstractDomainObject<Long> {
 
+    @Id
+    @GeneratedValue
+    private Long id;
+
```

Comment:
No need to add this - it is inherited from AbstractDomainObject<Long>.

Why did you feel you needed to add it?